idaes/generic_models/properties/core/generic/generic_property.py

```diff
@@ -1263,9 +1263,17 @@ def initialize(blk, state_args=None, state_vars_fixed=False,
                             blk[k].true_to_appr_species[p, j])
                     # Need to calculate all flows before doing mole fractions
                     for p, j in blk[k].params.apparent_phase_component_set:
-                        calculate_variable_from_constraint(
-                            blk[k].mole_frac_phase_comp_apparent[p, j],
-                            blk[k].appr_mole_frac_constraint[p, j])
+                        x = value(
+                            blk[k].flow_mol_phase_comp_apparent[p, j] /
+                            sum(blk[k].flow_mol_phase_comp_apparent[p, jj]
+                                for jj in blk[k].params.apparent_species_set))
+                        lb = blk[k].mole_frac_phase_comp_apparent[p, j]._lb
+                        if lb is not None and x <= lb:
+                            blk[k].mole_frac_phase_comp_apparent[
+                                p, j].set_value(lb)
```

Comment:
Why are you only snapping to the lower bound?  Should you not also check and snap to the upper bound?

Reply:
The upper bound _should_ not be an issue as there is some slack there (the bound is slightly greater than 1), so there should be no way the calculation returns a value greater than 1. Given this, I think it would actually be better not to have the check - that way if something ever does occur that causes a value greater than 1, we will start seeing warnings to tell us that something is off.